### PR TITLE
feat: add logging to geocoder endpoints

### PR DESCRIPTION
**Pull request by rosvik**
Moves the graphql logging to it's own function, `logResponse`, which then is reused for geocoder and vipps endpoints.

resolves https://github.com/AtB-AS/kundevendt/issues/3977

### CODE REVIEW

File: src/utils/fetch-client.ts [hunk 1]
```diff
@@ -10,26 +12,39 @@ const agent = new Agent({
...
     ...init,
     headers: {
       'ET-Client-Name': ET_CLIENT_NAME,
+      'X-Correlation-Id': headers['correlationId'] ?? '',
       ...init.headers,
     },
   });
+
+  logResponse({
+    message: init.method ?? 'unknown method',
```
Comment (rosvik) [inline src/utils/fetch-client.ts:29]: Is method (GET or POST) ok to use here?

File: src/utils/log.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  if (rateLimitUsed && rateLimitAllowed) {
+    let operationNameGroup;
+    if (url.includes('/mobility')) {
+      operationNameGroup = 'mobility';
+    } else if (url.includes('/journey-planner')) {
+      operationNameGroup =
+        operationName == 'Trips' ? 'planner-trip' : 'planner-nontrip';
+    } else if (url.includes('/geocoder')) {
+      operationNameGroup = 'geocoder';
```
Comment (rosvik) [inline src/utils/log.ts:33]: Not sure what to set operationNameGroup to here. Does 'geocoder' work?

**Comment by rosvik** [comment]
ref. https://github.com/AtB-AS/team-platform/issues/223

- [ ] `severity` - "INFO", "WARNING", or "ERROR"
- [ ] `time` - RFC3339-formatted timestamp
- [ ] `message` - a lower-cased string for what's being logged
  - for handlers use `handle request`
  - for outgoing HTTP requests use `http call`
  - for outgoing gRPC requests use `grpc call`
  - for outgoing GraphQL requests use `qraphql call`

For HTTP requests, add the following fields:

- [ ] `correlationId` - a UUID v4 formatted as a string, this should be created when the request is handled
- [ ] `method` - string of the HTTP method, e.g. "GET"
- [ ] `url` - the URI for the request (typically without the scheme/host)
- [ ] `code` - status code for the response
- [ ] `duration` - how long the request took to process

For external-facing services:

- [ ] `requestId` - contents of the HTTP header `Atb-Request-Id`
- [ ] `installId` - contents of the HTTP header `Atb-Install-Id`
- [ ] `appVersion` - contents of HTTP header `Atb-App-Version`

When available:

- [ ] `customerNumber` - customer number of the logged in user or the profile we get an event for
- [ ] `customerAccountId` - customer account id of the logged in user or the acount we get an event for

For rate limiting (when available):

- [ ] `rateLimitUsed` - value from `Rate-Limit-Used` header
- [ ] `rateLimitAllowed` - value from `Rate-Limit-Allowed` header

File: src/api/vipps-login/index.ts [hunk 3]
```diff
@@ -61,12 +63,25 @@ export default (server: Hapi.Server) => () => {
...
+          requestHeaders: h.request,
+          duration: vippsTimer.getElapsedMs(),
+        });
         if (phoneNumber) {
           const auth = app.auth();
           try {
+            const authTimer = new Timer();
             const user = await auth.getUserByPhoneNumber(`+${phoneNumber}`);
+            logResponse({
+              message: 'firebase call',
```
Comment (rosvik) [inline src/api/vipps-login/index.ts:80]: Should I use another message here?

File: src/api/vipps-login/index.ts [hunk 4]
```diff
@@ -61,12 +63,25 @@ export default (server: Hapi.Server) => () => {
         {nonce: nonce, state: state},
       );
       if (params.access_token) {
+        const vippsTimer = new Timer();
         const userInfo = await client.userinfo(params);
         const phoneNumber = userInfo.phone_number;
+        logResponse({
+          message: 'openid call',
```
Comment (rosvik) [inline src/api/vipps-login/index.ts:70]: Should I use another message here?

File: src/api/vipps-login/index.ts [hunk 5]
```diff
@@ -61,12 +63,25 @@ export default (server: Hapi.Server) => () => {
...
         if (phoneNumber) {
           const auth = app.auth();
           try {
+            const authTimer = new Timer();
             const user = await auth.getUserByPhoneNumber(`+${phoneNumber}`);
+            logResponse({
+              message: 'firebase call',
+              requestHeaders: h.request,
+              duration: authTimer.getElapsedMs(),
+              customerAccountId: user.uid,
```
Comment (rosvik) [inline src/api/vipps-login/index.ts:83]: Not sure about this, since I were unable to test it. Is `customerAccountId` = `uid` here?

Comment (mjenssen) [reply src/api/vipps-login/index.ts:83]: Not exactly sure, but if it's the Firebase Auth id, then no. It should be of the format `<codespace>:CustomerAccount:<id>`, e.g. for AtB we have `ATB:CustomerAccount:<id>`.

Comment (rosvik) [reply src/api/vipps-login/index.ts:83]: We'll see in staging, then 🤞

**Review by mjenssen — APPROVED** [review]
LGTM! 🚀

**Pull request merged**